### PR TITLE
SOHO-7935 Add findRowsByValue to datagrid control

### PR DESCRIPTION
**Pull request by micheal-cao**
> Explain the **details** for making this change. What existing problem does the pull request solve?
current datagrid control does not provide a function for find rows by field value.  this delivery add such function to  datagrid control.

sometimes, we need find rows numbers based on a row's field name, for example, we need find a row based on a row's unique id, sometimes, such id is internal, so we could not find it by going through all cells, also, performance is another consideration.

> **Related issue (required)**:
SOHO-7935 Add findRowsByValue to datagrid control

> **Steps necessary to review your pull request (required)**:
the logic is similar to the existing function selectedRows(), which selectedRows checks row's selection status, this function checks value of the field of data for rows.

> Other Details:
n/a

### CODE REVIEW

File: src/components/datagrid/datagrid.js [hunk 1]
```diff
@@ -5805,6 +5805,33 @@ Datagrid.prototype = {
     return selectedRows;
   },
 
+  /**
+   * Returns an array of row numbers for the rows containing the value for the field name.
+   */
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js]: Should add @params to the jsdoc

* @param {string}  [settings.fieldname] The field name to search
* @param {string}  [settings.value] The value to use in the search..

File: src/components/datagrid/datagrid.js [hunk 2]
```diff
@@ -5805,6 +5805,33 @@ Datagrid.prototype = {
...
+  findRowsByValue(fieldName, value) {
+    const s = this.settings;
+    const dataset = s.treeGrid ? s.treeDepth : s.dataset;
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js]: Does this work for tree grid? or untested?

**Comment by tmcconechy** [comment]
Im wondering if you saw the keywordSearch feature? if that can be re-utilized but i guess i dont have a problem with adding something like this.

File: src/components/datagrid/datagrid.js (hunk 2 above)
Comment (micheal-cao) [reply src/components/datagrid/datagrid.js]: Yes, It works for tree grid. I have tested in both regular table and tree table.

File: src/components/datagrid/datagrid.js (hunk 1 above)
Comment (micheal-cao) [reply src/components/datagrid/datagrid.js]: thanks, i would update it later

**Comment by micheal-cao** [comment]
keywordSearch searches cell text, but I hope that we could provide an ability of searching data model directly.

File: src/components/datagrid/datagrid.js (hunk 2 above)
Comment (tmcconechy) [reply src/components/datagrid/datagrid.js]: Yes i'll hit that soon when i go through those.

**Pull request merged**